server/v2/stf/stf.go

```diff
@@ -108,17 +104,63 @@ func (s STF[T]) validateTx(ctx context.Context, state store.WritableState, gasLi
 	return validateCtx.gasUsed, validateCtx.events, applyStateChanges(state, validateState)
 }
 
-func (s STF[T]) execTx(ctx context.Context, state store.WritableState, gasLimit uint64, tx T) (msgResp Type, gasUsed uint64, execEvents []event.Event, err error) {
+// execTx executes the tx messages on the provided state. If the tx fails then the state is discarded.
+func (s STF[T]) execTx(ctx context.Context, state store.WritableState, gasLimit uint64, tx T) (msgsResp []Type, gasUsed uint64, execEvents []event.Event, _err error) {
```

Comment:
nit: Can we avoid named returns on complex functions?

Reply:
addressed!